cpp/src/gandiva/gdv_string_function_stubs.cc

```diff
@@ -264,6 +264,356 @@ const char* gdv_fn_lower_utf8(int64_t context, const char* data, int32_t data_le
   return out;
 }
 
+//converts utf8 to utf16 encoding
+GANDIVA_EXPORT
+const char* gdv_fn_encode(int64_t context, const char* data, int32_t data_len,
+                          const char* charset, int32_t charset_len, int32_t* out_len) { 
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+  
+  char* out = reinterpret_cast<char*>(
+    gdv_fn_context_arena_malloc(context, 32 * data_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context,
+    "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  int32_t char_len;
+  uint32_t char_codepoint;
+  
+
+  if (memcmp(charset, "UTF-16BE", 8) == 0) {
```

Comment:
There is a way to compare it in a case-insensitive way? If the user passes `utf-16be` , will it work?